patches/minecraft/net/minecraft/world/entity/projectile/AbstractArrow.java.patch

```diff
@@ -9,14 +9,35 @@
           this.m_20095_();
        }
  
-@@ -186,7 +_,9 @@
+@@ -171,6 +_,7 @@
+             vec33 = hitresult.m_82450_();
+          }
+ 
++         entity_hit_check:
+          while(!this.m_213877_()) {
+             EntityHitResult entityhitresult = this.m_6351_(vec32, vec33);
+             if (entityhitresult != null) {
+@@ -186,9 +_,20 @@
                 }
              }
  
 -            if (hitresult != null && !flag) {
+-               this.m_6532_(hitresult);
+-               this.f_19812_ = true;
 +            if (hitresult != null && hitresult.m_6662_() != HitResult.Type.MISS && !flag) {
-+               if (net.minecraftforge.event.ForgeEventFactory.onProjectileImpact(this, hitresult))
-+                  break;
-                this.m_6532_(hitresult);
-                this.f_19812_ = true;
++               switch (net.minecraftforge.event.ForgeEventFactory.onProjectileImpactResult(this, hitresult)) {
++                  case CONTINUE: // No need to finish the loop when result is CONTINUE
++                     break entity_hit_check;
```

Comment:
We try to stay away from labeled loops as it is discouraged, and just a pain to read in my opinion, you can have the same effect by setting entityhitresult == null; Or not using a switch.